diff -Nur fribidi-0.10.4.orig/fribidi_mem.h fribidi-0.10.4/fribidi_mem.h
--- fribidi-0.10.4.orig/fribidi_mem.h	2004-10-25 07:18:45.000000000 +0100
+++ fribidi-0.10.4/fribidi_mem.h	2004-10-25 07:24:46.000000000 +0100
@@ -23,6 +23,7 @@
 #define FRIBIDI_MEM_H
 
 #include <stdlib.h>
+#include <stdint.h>
 
 #include "fribidi_config.h"
 
@@ -31,21 +32,16 @@
 {
 #endif
 
-#define FRIBIDI_INT8	char
-#define FRIBIDI_INT16	short
-#define FRIBIDI_INT32	long
-#define FRIBIDI_INT	int
-
   typedef int fribidi_boolean;
 
-  typedef signed FRIBIDI_INT8 fribidi_int8;
-  typedef unsigned FRIBIDI_INT8 fribidi_uint8;
-  typedef signed FRIBIDI_INT16 fribidi_int16;
-  typedef unsigned FRIBIDI_INT16 fribidi_uint16;
-  typedef signed FRIBIDI_INT32 fribidi_int32;
-  typedef unsigned FRIBIDI_INT32 fribidi_uint32;
-  typedef signed FRIBIDI_INT fribidi_int;
-  typedef unsigned FRIBIDI_INT fribidi_uint;
+  typedef int8_t fribidi_int8;
+  typedef uint8_t fribidi_uint8;
+  typedef int16_t fribidi_int16;
+  typedef uint16_t fribidi_uint16;
+  typedef int32_t fribidi_int32;
+  typedef uint32_t fribidi_uint32;
+  typedef int fribidi_int;
+  typedef unsigned int fribidi_uint;
 
 #define FRIBIDI_TRUE	1
 #define FRIBIDI_FALSE	0
